File: Algorithm_404_Sum of Left Leaves.cpp
#include <iostream>
using namespace std;
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int sum;
    int sumOfLeftLeaves(TreeNode* root) {
        sum=0;
        sumOfLeftLeaves(root,1);
        return sum;
    }
    void sumOfLeftLeaves(TreeNode* root, int lr)
    {
        if (!root) return;
        if (!root->left && !root->right && lr==0) sum+=root->val;
        if (root->left) sumOfLeftLeaves(root->left,0);
        if (root->right) sumOfLeftLeaves(root->right,1);
    }
};

File: Algorithm_404_Sum of Left Leaves.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {

    private int result = 0;

    public int sumOfLeftLeaves(TreeNode root) {
        preOrder(root, false); // the root itself is not considered as a leaf
        return result;
    }

    private void preOrder(TreeNode root, boolean goLeft) {
        if (root.left == null && root.right == null && goLeft) { // is left leaf
            result += root.val;
            return;
        }
        if (root.left != null) {
            preOrder(root.left, true);
        }
        if (root.right != null) {
            preOrder(root.right, false);
        }
    }
}